extensions/amp-a4a/0.1/a4a-variable-source.js

```diff
@@ -113,10 +112,58 @@ export class A4AVariableSource extends VariableSource {
       return getNavigationData(this.win_, 'redirectCount');
     });
 
+    this.set('HTML_ATTR',
+        /** @type {function(...*)} */(this.htmlAttrBinding_.bind(this)));
+
     for (let v = 0; v < WHITELISTED_VARIABLES.length; v++) {
       const varName = WHITELISTED_VARIABLES[v];
       const resolvers = this.globalVariableSource_.get(varName);
       this.set(varName, resolvers.sync).setAsync(varName, resolvers.async);
     }
   }
+
+  /**
+   * Provides a binding for getting attributes from the DOM.
+   * Most such bindings are provided in src/service/url-replacements-impl, but
+   * this one needs access to this.win_.document, which if the amp-analytics
+   * tag is contained within an amp-ad tag will NOT be the parent/publisher
+   * page. Hence the need to put it here.
+   * @param {string} cssSelector Elements matching this selector will be
+   *     included, provided they have at least one of the attributeNames
+   *     set, up to a max of 10. May be URI encoded.
+   * Note: Additional params will be the names of the attributes whose values
+   * will be returned. There should be at least 1.
+   * @returns {string}
+   */
+  htmlAttrBinding_(cssSelector) {
```

Comment:
Doc should say more about the format of the return value, since it's pretty complex.  It looks like it's a string that json decodes to a list of objects (one for each time the selector matched) mapping attributes to their values, with empty objects excluded?

Reply:
Can do, though currently return format is likely wrong, since a4avariablesource.js's initialize() method will call ```this.set(varName, resolvers.sync).setAsync(varName, resolvers.async);```. So I'll probably end up returning { sync: function() { return what I'm returning now }, async: return a promise that returns what I'm returning now }
Will document once it's correct.

Reply:
Done. (And my guess about needing to change the return value format was incorrect.)